pkg/rancher-desktop/main/commandServer/httpCommandServer.ts

```diff
@@ -630,6 +639,79 @@ export class HttpCommandServer {
       }
     }
   }
+
+  protected async listSnapshots(request: express.Request, response: express.Response, context: commandContext): Promise<void> {
+    const snapshots = await this.commandWorker.listSnapshots(context);
+
+    response.status(200).type('json').send(snapshots);
+  }
+
+  protected async createSnapshot(request: express.Request, response: express.Response, context: commandContext): Promise<void> {
+    try {
+      const [data, payloadError] = await serverHelper.getRequestBody(request, MAX_REQUEST_BODY_LENGTH);
+
+      if (payloadError) {
+        response.status(400).type('txt').send('The snapshot is invalid');
```

Comment:
Did you need an early return here?

Reply:
The early return here will catch errors caused by body size exceeds.

Reply:
But it's _not_ returning; it will continue to try to do a second `response.status()` on line 702 (or go create a snapshot on line 704)?

Reply:
Completely missed the sense of your comment. Yes, a return is need here.